### PR TITLE
plugin/federation: Add upstream option to federation

### PR DESCRIPTION
**Pull request by chrisohaver**
### 1. Why is this pull request needed and what does it do?
Adds upstream option to federation, so that answers for federated services (which are rendered as CNAME records) can include the records they point to.

### 3. Which documentation changes (if any) need to be made?

### CODE REVIEW

**Comment by chrisohaver** [comment]
I added a CI test for this - coredns/ci#72.  I don't think a unit test for upstream function is easy.

**Comment by chrisohaver** [comment]
FYI @shashidharatd

File: plugin/federation/federation.go [hunk 1]
```diff
@@ -105,6 +108,15 @@ func (f *Federation) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.
 
 	m.Answer = []dns.RR{service.NewCNAME(state.QName(), service.Host)}
 
+	if f.Upstream != nil {
+		aRecord, err := f.Upstream.Lookup(state, service.Host, state.QType())
+		if err != nil {
+		}
```
Comment (fturib) [inline plugin/federation/federation.go]: What is the usage of this test with empty action ?

**Review by fturib — COMMENTED** [review]
LGTM, besides one comment that seems to be related to dead code.

**Review by fturib — APPROVED** [review]

**Pull request merged**